### PR TITLE
Registration and Authentication

### PR DESCRIPTION
**Pull request by cindyzhang12**
## Summary

- create registration page
- integrated authentication
- still have to do:
   - errors on form submission
   - submit for auth page

### CODE REVIEW

File: src/components/pages/Authentication/Authentication.test.tsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import { Authentication } from './Authentication';
+
+describe('Tooltip test', () => {
```
Comment (kurund) [inline src/components/pages/Authentication/Authentication.test.tsx:5]: Please fix the description

File: src/components/pages/Authentication/Authentication.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  location: any;
+}
+
+interface State {
+  authCode: string;
+  authResponse: string;
+  sentAuth: boolean;
+}
+
+const useStyles = makeStyles((theme: Theme) =>
```
Comment (kurund) [inline src/components/pages/Authentication/Authentication.tsx:21]: Can you move this code to css file.

File: src/components/pages/Registration/Registration.tsx [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+import VisibilityOff from '@material-ui/icons/VisibilityOff';
+import Button from '@material-ui/core/Button';
+import { Redirect } from 'react-router-dom';
+import clsx from 'clsx';
+import axios from 'axios';
+
+export interface RegistrationProps {}
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
```
Comment (kurund) [inline src/components/pages/Registration/Registration.tsx:20]: Move this to css file

File: src/components/pages/Registration/Registration.tsx [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+  authRedirect: boolean;
+  authToken: any;
+  authCode: any;
+  errors: any;
+}
+
+export const Registration: React.SFC<RegistrationProps> = () => {
+  const classes = useStyles();
+
+  const [values, setValues] = React.useState<State>({
```
Comment (kurund) [inline src/components/pages/Registration/Registration.tsx:49]: Please follow useState() convention

File: src/components/pages/Registration/Registration.tsx [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+    event.preventDefault();
+  };
+
+  const handleMouseDownConfirmPassword = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+  };
+
+  const handleSubmit = () => {
+    axios
+      .post('http://localhost:4000/api/v1/registration', {
```
Comment (kurund) [inline src/components/pages/Registration/Registration.tsx:84]: We should not hardcode the URL here. Please use: REACT_APP_GLIFIC_API

File: src/components/pages/Registration/Registration.tsx [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+          password: values.password,
+          password_confirmation: values.confirmPassword,
+        },
+      })
+      .then(function (response: any) {
+        console.log(response);
+        const responseString = JSON.stringify(response);
+        setValues((values) => ({ ...values, authToken: responseString }));
+        axios
+          .post('http://localhost:4000/api/v1/registration/send_otp', {
```
Comment (kurund) [inline src/components/pages/Registration/Registration.tsx:97]: Same as above

File: src/components/pages/Authentication/Authentication.tsx [hunk 7]
```diff
@@ -0,0 +1,55 @@
+import React, { useState } from 'react';
```
[the remark this answers is not in the document]
Comment (cindyzhang12) [reply src/components/pages/Authentication/Authentication.tsx:1]: Good point! I'll change it to ConfirmOTP.tsx :)

File: src/config/axios.ts [hunk 8]
```diff
@@ -0,0 +1,5 @@
+export const REACT_APP_GLIFIC_REGISTRATION_API =
+  process.env.REACT_APP_GLIFIC_API + '/v1/registration';
+export const REACT_APP_GLIFIC_AUTHENTICATION_API =
+  process.env.REACT_APP_GLIFIC_API + '/v1/registration/send_otp';
+export const REACT_APP_GLIFIC_CHAT_REDIRECT = 'http://localhost:3000/chat';
```
Comment (rathorevaibhav) [inline src/config/axios.ts:5]: let's update this too. Something like:
```suggestion
export const REACT_APP_GLIFIC_CHAT_REDIRECT = process.env.REACT_APP_GLIFIC_URL . '/chat';
```

File: src/components/pages/Registration/Registration.tsx [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+
+export const Registration: React.SFC<RegistrationProps> = () => {
+  const [password, setPassword] = useState('');
+  const [userName, setUserName] = useState('');
+  const [phoneNumber, setphoneNumber] = useState('');
+  const [confirmPassword, setconfirmPassword] = useState('');
+  const [showPassword, setshowPassword] = useState(false);
+  const [showConfirmPassword, setshowConfirmPassword] = useState(false);
+  const [authToken, setauthToken] = useState('');
+  const [authCode, setauthCode] = useState();
```
Comment (rathorevaibhav) [inline src/components/pages/Registration/Registration.tsx:31]: minor thing but can we also update the other methods here? `setPhoneNumber`, `setConfirmPassword`, `setShowPassword` etc...

File: src/components/pages/ConfirmOTP/ConfirmOTP.tsx [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+import { Button } from '../../UI/Form/Button/Button';
+import { REACT_APP_GLIFIC_CHAT_REDIRECT } from '../../../config/axios';
+import clsx from 'clsx';
+
+export interface ConfirmOTPProps {
+  location: any;
+}
+
+export const ConfirmOTP: React.SFC<ConfirmOTPProps> = (props) => {
+  const [userAuthCode, setuserAuthCode] = useState('');
```
Comment (rathorevaibhav) [inline src/components/pages/ConfirmOTP/ConfirmOTP.tsx:16]:
```suggestion
  const [userAuthCode, setUserAuthCode] = useState('');
```

**Pull request closed without merging**